### DISCUSSION TITLE
Possibility to save the session id or a token to keep the past conversation in history (Like chatGPT)

### DISCUSSION THREAD

**Discussion by CarinaSchoppe** [openai/openai-python#1225] [Ideas]
I´d like to know, if there is a feature that allowes me to utilize the API to include  something like a session id or token or someelse thats capable of keeping the past and latest conversation. Just like ChatGPT.

Or should the GPT Assistens be used for that.

My IDEA: Feed the AI 10 complete books time after time and than ask questions related to that topic.
(These 10 books or more would exeed the 128k tokenlimit of gpt4 turbo prev.)
Thoughout my jurney with openai I´d love to see something implemented like that.

**Comment by rattrayalex** [comment]
The Assistants API is intended for this: https://platform.openai.com/docs/assistants/overview
